frappe/database/query.py

```diff
@@ -165,9 +165,7 @@ def _apply_filter(
 					(table.parent == self.table.name) & (table.parenttype == self.doctype)
 				)
 
-		if isinstance(_value, (str, int)):
-			_value = str(_value)
```

Comment:
This was done to handle primary keys IIRC 

@phot0n

Reply:
The generated query is more accurate but the results are the same. Is there anything I am missing?
```py
frappe.qb.get_query("GP Discussion", filters={'name': 1})
SELECT `name` FROM `tabGP Discussion` WHERE `name`=1

frappe.qb.get_query("GP Discussion", filters={'name': "1"})
SELECT `name` FROM `tabGP Discussion` WHERE `name`='1'
```

Reply:
Yep. Seems to be compatible with postgres too.